shm/Store.cpp

```diff
@@ -1,33 +1,152 @@
 #include "Store.hpp"
 
 #include <algorithm>
+#include <random>
 
-Store::Store(std::shared_ptr<Time>& time)
-    : time_(time) {
-    time_->addObserver(this);
+#include "Alcohol.hpp"
+#include "Fruit.hpp"
+#include "Item.hpp"
+
+Store::Store(std::shared_ptr<Time>& time) {
+    time_ = time;
+    time->addObserver(this);
+    market_ = makeStock(generateFruits(), generateAlcos(), generateItems());
+}
+
+size_t Store::generateRandom(int min, int max) const {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distirb(min, max);
+
+    return static_cast<size_t>(distirb(gen));
+}
+
+Cargos Store::generateFruits() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Fruit fruit(fruitNames[generateRandom(0, 5)], generateRandom(1, 20), generateRandom(10, 30), generateRandom(1, 10));
+        if(std::none_of(begin(result), end(result),[&fruit](const auto& ptr){ return ptr->getName() == fruit.getName();})) {
+            result.emplace_back(std::make_shared<Fruit>(fruit));
+            i++;
+        }
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
+
+Cargos Store::generateAlcos() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Alcohol alco(alcoNames[generateRandom(0, 5)], generateRandom(1, 5), generateRandom(30, 100), generateRandom(40, 96));
+        if(std::none_of(begin(result), end(result),[&alco](const auto& ptr){ return ptr->getName() == alco.getName();})) {
+            result.emplace_back(std::make_shared<Alcohol>(alco));
+            i++;
+        }
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
+
+Cargos Store::generateItems() const {
+    Cargos result;
+    result.reserve(marketSection);
+
+    size_t i = 0;
+    while(i < marketSection) {
+        Item item(itemNames[generateRandom(0, 5)], generateRandom(1, 10), generateRandom(30, 100), Rarity::rare);
+        if(std::none_of(begin(result), end(result),[&item](const auto& ptr){ return ptr->getName() == item.getName();})) {
+            result.emplace_back(std::make_shared<Item>(item));
+            i++;
+        }
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
+
+Cargos Store::makeStock(const Cargos& fruits, const Cargos& alcos, const Cargos& items) {
+    Cargos result;
+    result.reserve(marketSection * 3);
+    for (size_t i = 0; i < marketSection; i++) {
+        result.push_back(fruits[i]);
+        result.push_back(alcos[i]);
+        result.push_back(items[i]);
+    }
+    result.shrink_to_fit();
+
+    return result;
+}
```

Comment:
If you call `generateFruits()`, `generateAlcos()`, etc. inside `makeStock` instead of providing fruits, alcos and items as parameters, then you can have  something like this:
```cpp
void Store::generateCargo() { 
    generateAlcohol();
    generateFruits();
    generateItems();
}
```

and then each `generateSomethin()` method can emplace_back directly to CargoStock of this Store. I think we would have less data copies between functions. 

I've done something like that on my "crazyShopping" branch.